File: java/dsframework/src/main/java/dsf/pool/ClientSourceException.java
package dsf.pool;

import dsf.register.MsgSvcAddr;

/**
 *
 * @author arksea
 */
public class ClientSourceException extends RuntimeException {

    private String regname;
    private MsgSvcAddr svcAddr;

    public ClientSourceException(String msg, String regname, MsgSvcAddr addr) {
        super(msg);
        this.regname = regname;
        this.svcAddr = addr;
    }

    public ClientSourceException(String msg, String regname, MsgSvcAddr addr, Throwable ex) {
        super(msg, ex);
        this.regname = regname;
        this.svcAddr = addr;
    }

    public String getRegname() {
        return regname;
    }

    public MsgSvcAddr getSvcAddr() {
        return svcAddr;
    }
}

File: csharp/dsf/dsf/src/pool/ClientSourceException.cs
using System;

namespace com.baidu.dsf.pool
{

    using MsgSvcAddr = com.baidu.dsf.register.MsgSvcAddr;

    /// 
    /// <summary>
    /// @author arksea
    /// </summary>
    public class ClientSourceException : Exception
    {

        private string regname;
        private MsgSvcAddr svcAddr;

        public ClientSourceException(string msg, string regname, MsgSvcAddr addr) : base(msg)
        {
            this.regname = regname;
            this.svcAddr = addr;
        }

        public ClientSourceException(string msg, string regname, MsgSvcAddr addr, Exception ex) : base(msg, ex)
        {
            this.regname = regname;
            this.svcAddr = addr;
        }

        public virtual string Regname
        {
            get
            {
                return regname;
            }
        }

        public virtual MsgSvcAddr SvcAddr
        {
            get
            {
                return svcAddr;
            }
        }
    }

}